### ISSUE TITLE
[google_maps_flutter] GoogleMap widget is black on android when not using mapsImplementation.useAndroidViewSurface = true

### ISSUE THREAD

**Issue by ziyad-aljohani** [flutter/flutter#147116]
### Steps to reproduce

Hello 
I use google maps flutter package, since 2 years, but now the map doesn't work well on some of Android devices, it's only a black widget.
I solved the problem by using
```
final GoogleMapsFlutterPlatform mapsImplementation =
      GoogleMapsFlutterPlatform.instance;
  if (mapsImplementation is GoogleMapsFlutterAndroid) {
    // Force Hybrid Composition mode.
    mapsImplementation.useAndroidViewSurface = true;
  }
```
But I think it impacts the performance of the map

### Expected results

It should work well like previously

### Actual results

Black empty widget.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:google_maps_flutter/google_maps_flutter.dart';
import 'package:flutter/material.dart';
import 'package:google_maps_flutter_android/google_maps_flutter_android.dart';
import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';

void main() async {
  //*****ENABLE this line 
  // _googleMapsFlutterSetup();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: GoogleMapPage(),
    );
  }
}

void _googleMapsFlutterSetup() {
  final mapsImplementation = GoogleMapsFlutterPlatform.instance;
  if (mapsImplementation is GoogleMapsFlutterAndroid) {
    mapsImplementation.useAndroidViewSurface = true;
  }
}

class GoogleMapPage extends StatefulWidget {
  GoogleMapPage({Key? key}) : super(key: key);

  @override
  State<GoogleMapPage> createState() => _GoogleMapPageState();
}

class _GoogleMapPageState extends State<GoogleMapPage> {
  @override
  Widget build(BuildContext context) {
    return GoogleMap(
      initialCameraPosition:
          const CameraPosition(target: LatLng(0.0, 0.0), zoom: 1),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel beta, 3.22.0-0.1.pre, on macOS 14.4 23E214 darwin-arm64, locale en-SA)
    • Flutter version 3.22.0-0.1.pre on channel beta at /Users/ziyad/Downloads/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 29babcb32a (2 weeks ago), 2024-04-03 17:17:04 -0500
    • Engine revision 97550907b7
    • Dart version 3.4.0 (build 3.4.0-282.1.beta)
    • DevTools version 2.34.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/XX/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.
```

</details>

**Comment by ziyad-aljohani** [comment]
The problem is from Flutter version 3.22.0-0.1.pre on channel beta.
I moved back to the stable channel and now it works well.

I won't use flutter channel beta on my app again.

**Comment by danagbemava-nc** [comment]
Hi @ziyad-aljohani, what device(s) are you seeing this issue on? The beta channel is currently at `3.22.0-0.2.pre`, do you still experience this with that version?

**Comment by ziyad-aljohani** [comment]
@danagbemava-nc 
Hello, happens on many android devices like
Huawei mate 20 x
Oneplus 7 pro
And many android phones
Also happens in 
7 WSVGA (Tablet) API 29 mobile emulator 

The new version of beta i didn’t try it

**Comment by danagbemava-nc** [comment]
Hi @ziyad-aljohani, I just checked on my Pixel with beta `3.22.0-0.1.pre` & `3.22.0-0.2.pre` but I could not reproduce the issue. 

What versions of android were the devices running? Did you get this issue in debug mode or was it in release mode? Did this happen after the app was backgrounded?

**Comment by danagbemava-nc** [comment]
![Screenshot from 2024-04-24 07-48-10](https://github.com/flutter/flutter/assets/88313112/02ece6dd-298d-431b-b859-cc4ecfafa90e)
![Screenshot from 2024-04-24 07-50-08](https://github.com/flutter/flutter/assets/88313112/08b1ca68-6ca3-431a-b32b-c59d1f69b5db)

It appears that this has been fixed on 3.22.0-0.2.pre. See the screenshots above.
Per our triage policy, I'll be closing this since it has been fixed in a newer version. 

Thank you
